OS/DiskOS/terminal.lua

```diff
@@ -419,6 +445,25 @@ function term.loop() --Enter the while loop of the terminal
           buffer = buffer:sub(1,inputPos-1)..paste..buffer:sub(inputPos,-1)
           inputPos = inputPos + paste:len()
         end
+      elseif a == "tab" then
+        --If the autocomplete suggestions list is empty, create a new one
+        if next(autocompleteSuggestions) == nil then
+          autocompleteSuggestions = term.autocomplete(buffer, commands)
+          suggestionIndex
+         = 1
+        --Otherwise, go to the next suggestion in the list (looping to the
+        --first if the end of the list is reached)
+        else
+          term.clearSuggestion(autocompleteSuggestions[suggestionIndex])
+          suggestionIndex = suggestionIndex + 1
+          if autocompleteSuggestions[suggestionIndex] == nil then
```

Comment:
This expression could be simplified:
 Values in acts as booleans:
- false, nil -> false.
- true, number, table, string, userdata -> true.

So this expression could be written as:
```lua
if not next(autocompleteSuggestions) then
```